### DISCUSSION TITLE
Running Netket with MPI

### DISCUSSION THREAD

**Discussion by tortijazz** [netket/netket#1367] [Q&A]
Hello everyone,

I am currently working on optimising a very large many-body quantum system, and I want to use MPI parallelisation. However, I must confess that I don't know how to use MPI, I've checked some tutorials and I'm the learning process, so I don't fully understand it and I'm confused. My question is:

Should I add the following lines to my_script.py

```
from mpi4py import MPI

comm = MPI.COMM_WORLD
size = comm.Get_size()
rank = comm.Get_rank()
```
and run on terminal `mpirun -n #processes python my_script.py` or just run `mpirun -n #processes python my_script.py` without the above four lines into my_script.py?

Best regards!

**Comment by PhilipVinc** [comment]
The lines above do nothing. If you have installed the MPI dependencies (mpi4jax, mpi4py) then running `mpirun -n #processes python script.py` will use MPI to distribute samples across different ranks.

Beware that our parallelisation strategy is performant only if you have many samples.
